core/player/src/player.ts

```diff
@@ -87,12 +87,41 @@ export class Player {
     commit: COMMIT,
   };
 
-  public readonly logger = new TapableLogger();
-  public readonly constantsController = new ConstantsController();
+  public readonly logger: TapableLogger = new TapableLogger();
+  public readonly constantsController: ConstantsController =
+    new ConstantsController();
   private config: PlayerConfigOptions;
   private state: PlayerFlowState = NOT_STARTED_STATE;
 
-  public readonly hooks = {
+  public readonly hooks: {
+    /** The hook that fires every time we create a new flowController (a new Content blob is passed in) */
```

Comment:
Remove redundant TS docs from actual object since it's captured in the explicit type now.